### ISSUE TITLE
[Proposal] Add the option "qpdf --help=examples"

### ISSUE THREAD

**Issue by rubyFeedback** [qpdf/qpdf#913]
It is here proposed to add this option:

    qpdf --help=examples

What shall this option do?

This option just shows some commandline examples. Something, say,
between 5-10 or something like that. I can't give an exact number; I 
think when you agree to the usefulness of this, you can decide what the
ideal number may be.

Of course I may say 100 examples. :D But ... I think this may be too
spammy and too much, so I just would recommend to focus on the
PRIMARY use cases, of, say, 100 different people. Their top 5 or top
ten or perhaps top 15 if we need more.

Ok. So .. what shall this show?

Well, this should typically show AND BRIEFLY explain some use cases
for qpdf ON THE COMMANDLINE.

Just the typical use cases for qpdf. Perhaps one rotate example. Perhaps
one example to merge/remove/split pages from a .pdf file. Or combine
two different .pdf files.

The format of this is not so important, but I would recommend to keep
it simple and succinct. And then perhaps also add a disclaimer where
to find MORE explanations and MORE examples, be it in any doc/
subdirectory or github-wiki page or some other location, for people
to want to read up more.

Use case and rationale for this proposal: I believe this may help first
time users with qpdf a LOT if the commandline version can quickly
show them some useful things, perhaps even a few tricks they did
not know yet. Perhaps also one more advanced and involved example,
but I would recommend to focus on SIMPLE use cases, up to a maximum
of MEDIUM use cases, and leave the complicated one to the more
advanced users instead.

**Comment by jberkenbilt** [comment]
Various suggestions have been made around the area of having command-line examples. I think this is a good way to do it. I'm not committing to a timeline, but I think this could be made to work.

**Comment by dreua** [comment]
Do you know tldr? It is available as CLI similar to `man` ad well as on the internet and specializes in examples. There are already a few for qpdf:

https://github.com/tldr-pages/tldr/blob/main/pages/common/qpdf.md?plain=1

**Comment by jberkenbilt** [comment]
@dreua Nope, that's new to me. I expect to have a chunk of qpdf time hopefully as soon as the week after next. I'll go through some of the easier tickets and knock them out, release 11.7, and then hopefully dive into my significant feature work. I doubt it will take me very long to generate a man page from the existing help framework, and it will have the advantage of being complete and consistent with the rest of the docs.

**Comment by dreua** [comment]
My recommendation of tldr was meant for @rubyFeedback as they seem to like cli examples :wink:

Still a good idea to have them in the pikepdf documentation / man page.
